common/models/role.js

```diff
@@ -289,7 +299,11 @@ module.exports = function(Role) {
     var resolver = Role.resolvers[role];
     if (resolver) {
       debug('Custom resolver found for role %s', role);
-      resolver(role, context, callback);
+
+      var promise = resolver(role, context, callback);
+      if (promise instanceof Promise) {
```

Comment:
This will not work when `resolver` uses a different promise implementation. Promises were designed for interoperability, basically the only requirement is to provide a `then` method. See https://github.com/strongloop/loopback/issues/1612#issuecomment-214748981

```
      if (result && typeof result.then === 'function') {
        result.then(function() { next(); }, next);
      }
```